controller_manager/src/controller_manager.cpp

```diff
@@ -541,34 +541,38 @@ bool ControllerManager::shutdown_controllers()
 
 void ControllerManager::init_controller_manager()
 {
-  controller_manager_activity_publisher_ =
-    create_publisher<controller_manager_msgs::msg::ControllerManagerActivity>(
-      "~/activity", rclcpp::QoS(1).reliable().transient_local());
-  rt_controllers_wrapper_.set_on_switch_callback(
-    std::bind(&ControllerManager::publish_activity, this));
-  resource_manager_->set_on_component_state_switch_callback(
-    std::bind(&ControllerManager::publish_activity, this));
+  if (!resource_manager_ && !robot_description_.empty())
+  {
+    init_resource_manager(robot_description_);
+  }
 
-  // Get parameters needed for RT "update" loop to work
-  if (is_resource_manager_initialized())
+  if (!is_resource_manager_initialized())
   {
-    if (params_->enforce_command_limits)
+    // fallback state
+    resource_manager_ = std::make_unique<hardware_interface::ResourceManager>(trigger_clock_, get_logger());
+    if (!robot_description_notification_timer_)
     {
-      resource_manager_->import_joint_limiters(robot_description_);
+      robot_description_notification_timer_ = create_wall_timer(
+        std::chrono::seconds(1),
+        [&]()
+        {
+          RCLCPP_WARN(get_logger(), "Waiting for data on 'robot_description' topic to finish initialization");
+        });
     }
-    init_services();
-  }
-  else
+  }else
   {
-    robot_description_notification_timer_ = create_wall_timer(
-      std::chrono::seconds(1),
-      [&]()
-      {
-        RCLCPP_WARN(
-          get_logger(), "Waiting for data on 'robot_description' topic to finish initialization");
-      });
+    RCLCPP_INFO(get_logger(),
+      "Resource Manager has been successfully initialized. Starting Controller Manager "
+      "services...");
```

Comment:
```suggestion
      "Resource Manager successfully initialized. Starting Controller Manager services...");
```

I know this is just being picky. I saved one line! :D